### DISCUSSION TITLE
Superset 3.0 deprecated API removal

### DISCUSSION THREAD

**Discussion by dpgaspar** [apache/superset#24332] [General]
# Superset 3.0 deprecated API removal

## Introduction

We have been migrating the now considered old API that lives on `/superset` resource name to the new API v1 that lives on `/api/v1/<resource>/`

This effort began with the initial API v1 proposal and CRUD server side rendered views replacement by React and move to SPA:

And has been pushed further with the goal of eliminating technical debt and reduce security risks.

Proposal for code removal of old API endpoints on `/superset` 

|PR| Old path | New Path | Old Method | FE references? | Can be removed? | Notes |
|--| -- | -- | -- | -- | -- | -- |
|✅ #24400| /recent_activity/<int:user_id>/ | /api/v1/log/recent_activity/<user_id>/ | GET | No | Yes | ENABLE_BROAD_ACTIVITY_ACCESS should we remove the user_id? |
|✅ #24333| /datasources/ | api/v1/dataset/ | GET | No | Yes |  |
|✅ #24266| /override_role_permissions/ | - | POST | No | Yes |  |
|✅ #24266| /request_access/ | - | POST
|✅ #24266| /approve | - | POST
|✅ #24423| /slice_json/<int:slice_id> | /api/v1/chart/<int:id>/data/ | GET → POST | No | Do we have legacy charts that use it? |  |
|✅ #24423| /annotation_json/<int:layer_id> | /api/v1/chart/<int:id>/data/ | GET | No | Do we have legacy charts that use it? |  |
|✅ #24335| /filter/<datasource_type>/<int:datasource_id>/<column>/ | /api/v1/datasource/<datasource_type>/ | GET | No | Yes |  |
|✅ #24342| /tables/<int:db_id>/<schema>/ | api/v1/database/<int:pk>/tables/ | GET | No | Yes |  |
|✅ #24353| /copy_dash/<int:dashboard_id>/ | api/v1/dashboard/<dash_id>/copy/ | POST | No | Yes |  |
|✅ #24353| /save_dash/<int:dashboard_id>/ | - | GET+POST | No | Yes |  |
|✅ #24353| /add_slices/<int:dashboard_id>/ | api/v1/chart/<id>/ | POST→PUT | No | Yes | Work in progress |
|✅ #24354| /testconn | /api/v1/database/test_connection/ | POST | No | Yes |  |
|✅ #24381| /available_domains/ | /api/v1/available_domains/ | GET | No | Yes |  |
|✅ #24400| /fave_dashboards_by_username/<username>/ | api/v1/dashboard/favorite_status/ | GET | No | Yes | ENABLE_BROAD_ACTIVITY_ACCESS New API does not use it |
|✅ #24400| /fave_dashboards/<int:user_id>/ | api/v1/dashboard/favorite_status/ | GET | No | Yes | ENABLE_BROAD_ACTIVITY_ACCESS New API does not use it |
|✅ #24400| /created_dashboards/<int:user_id>/ | api/v1/dashboard/ | GET | No | Yes | ENABLE_BROAD_ACTIVITY_ACCESS New API does not use it |
|✅ #24400| /user_slices | api/v1/chart | GET | No | Yes | ENABLE_BROAD_ACTIVITY_ACCESS New API does not use it |
|✅ #24400| /created_slices/<int:user_id>/ | api/v1/chart | GET | No | Yes | ENABLE_BROAD_ACTIVITY_ACCESS New API does not use it |
|✅ #24400| /fave_slices/<int:user_id>/ | api/v1/chart | GET |  |  | ENABLE_BROAD_ACTIVITY_ACCESS New API does not use it |
|✅ #24400| /favstar/<class_name>/<int:obj_id>/<action>/ | api/v1/dashboard, api/v1/chart/<pk>/favorites/ | POST DELETE | No | Yes |  |
|✅ #24375| /get_or_create_table/ | api/v1/dataset/get_or_create/ | POST | No | Yes |  |
|✅ #24375| /sqllab_viz/ | api/v1/dataset/ | POST | No | Yes |  |
|✅ #24354| /extra_table_metadata/<int:database_id>/<table_name>/<schema>/ | api/v1/database/<int:pk>/table_extra/<table_name>/<schema_name>/ | GET | No | Yes |  |
|✅ #24359| /estimate_query_cost/<int:database_id>/ | /api/v1/sqllab/estimate/ | POST | No | Yes |  |
|✅ #24359| /results/<key>/ | /api/v1/sqllab/results/ | GET | No | Yes |  |
|✅ #24360| /stop_query/ | /api/v1/query/stop | PUT | No | Yes |  |
|✅ #24354| /validate_sql_json/ | api/v1/database/<pk>/validate_sql/ | POST | No | Yes |  |
|✅ #24359| /sql_json/ | api/v1/sqllab/execute/ | POST | No | Yes |  |
|✅ #24359| /csv/<client_id> | api/v1/sqllab/export/ | GET | No | Yes |  |
|-| /fetch_datasource_metadata | api/v1/database/<int:pk>/table/<table_name>/<schema_name>/ | GET | Yes | No |  |
|✅ #24360| /queries/<float:last_updated_ms> | api/v1/query/updated_since | GET | No | Yes |  |
|✅ #24360| /search_queries | api/v1/query/ | GET | No | Yes |  |
|✅ #24354| /schemas_access_for_file_upload | /api/v1/database/{pk}/schemas_access_for_file_upload/ | GET | No | Yes |  |
|  |  |  |  |  |  |  |

Non migrated endpoints that still live on `/superset`

| Old path | Type | Note |
| --- | --- | --- |
| /slice/<int:slice_id>/ | Server side rendered | Related with SPA project |
| /explore_json/data/<cache_key> | “REST” |  |
| /explore_json/ | “REST” | Pending by legacy charts |
| /import_dashboards/ | Server side rendered |  |
| /warm_up_cache/ | “REST” | Work in progress |
| /dashboard/<dashboard_id_or_slug>/ | Server side rendered | related with SPA project |
| /dashboard/p/<key>/ | Server side rendered | related with SPA project |
| /log/ | “REST” | Need to check |
| /theme/ | Server side rendered |  |
| /welcome/ | Server side rendered | related with SPA project |
| /profile/<username>/ | Server side rendered | related with SPA project |
| /sqllab/ | Server side rendered | related with SPA project |
| /sqllab/history/ | Server side rendered | related with SPA project |
| /explore | Server side rendered | related with SPA project |

**Comment by michael-s-molina** [comment]
This is awesome @dpgaspar! Thank you very much 🙏🏼

**Comment by michael-s-molina** [comment]
@john-bodley @villebro
